config/initializers/prometheus.rb

```diff
@@ -1,18 +1,39 @@
-if Settings.prometheus_exporter_host.present? && !$0.include?('prometheus_exporter')
+unless Rails.env.test?
+  require 'prometheus_exporter'
   require 'prometheus_exporter/client'
   require 'prometheus_exporter/metric'
   require 'prometheus_exporter/middleware'
   require 'prometheus_exporter/instrumentation'
-  client = PrometheusExporter::Client.new(
-    host: Settings.prometheus_exporter_host,
-    port: Settings.prometheus_exporter_port
-    # Add custom_labels for app_name here (consumer and webserver)
-  )
+
+  def ensure_exporter_server
+    require 'socket'
+    TCPSocket.open("127.0.0.1", Settings.metrics_port) {}
+  rescue Errno::ECONNREFUSED
+    require 'prometheus_exporter/server'
+
+    server = PrometheusExporter::Server::WebServer.new(port: Settings.metrics_port)
+    server.start
```

Comment:
Consider adding another rescue in case two processes try to start the server at the same time.